components/automate-gateway/gateway/services.go

```diff
@@ -673,6 +684,84 @@ func (s *Server) configMgmtReportExportHandler(w http.ResponseWriter, r *http.Re
 	}
 }
 
+func (s *Server) DeploymentStatusHandler(w http.ResponseWriter, r *http.Request) {
+	const (
+		actionV1   = "read"
+		resourceV1 = "service_info:status"
+		actionV2   = "system:status:get"
+		resourceV2 = "system:service:status"
+	)
+
+	ctx, err := s.authRequest(r, resourceV1, actionV1, resourceV2, actionV2)
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusForbidden)
+		return
+	}
+
+	deploymentClient, err := s.clientsFactory.DeploymentServiceClient()
+	if err != nil {
+		http.Error(w, "grpc service for deployment service unavailable", http.StatusServiceUnavailable)
+		return
+	}
+
+	status, err := deploymentClient.Status(ctx, &deploy_api.StatusRequest{})
+	if err != nil {
+		http.Error(w, err.Error(), http.StatusInternalServerError)
+		return
+	}
+
+	serviceStatus := make([]ServiceStatus, len(status.ServiceStatus.Services))
+
+	var overallOks bool = true
+	for index, svc := range status.ServiceStatus.Services {
+		overallOks = overallOks && (svc.State == deploy_api.ServiceState_OK)
+
+		serviceStatus[index] = ServiceStatus{
+			Service: svc.Name,
+			Status:  svc.State.String(),
+		}
+	}
+
+	result := DeploymentStatus{
+		Ok:              overallOks,
+		ServiceStatuses: serviceStatus,
+	}
+
+	// Because this is a monitoring endpoint, we want to return 500 if services are down.
+	// This is consistent with Chef Infra Server's behavior and a nicer experience for users
+	// running monitoring software with built-in support for http checks. (e.g. nagios)
+	if overallOks {
+		w.WriteHeader(http.StatusOK)
+	} else {
+		w.WriteHeader(http.StatusInternalServerError)
+	}
+
+	// Provide prettyprinted output.
+	var jsonBody []byte
+	if _, ok := r.URL.Query()["pretty"]; ok {
+		w.Header().Set("Content-Type", "application/json+pretty")
```

Comment:
Should the content-type be "application/json"? I wasn't sure. ("application/json+pretty" is the accept header for negotiating with grpc-gateway itself.)

Reply:
My guess is we are find using application/json for both.